internal/pkg/utils/deepcopy/deepcopy.go

```diff
@@ -29,21 +29,25 @@ func CopyTranslateSteps(value interface{}, callback TranslateFunc, steps Steps,
 	return clone.Interface()
 }
 
-type VisitedMap map[uintptr]bool
+type VisitedPtrMap map[uintptr]*reflect.Value
+
+type CloneNestedFn func(clone reflect.Value)
 
 // translateRecursive is modified version of https://gist.github.com/hvoecking/10772475
-func translateRecursive(clone, original reflect.Value, callback TranslateFunc, steps Steps, visited VisitedMap) {
+func translateRecursive(clone, original reflect.Value, callback TranslateFunc, steps Steps, visitedPtr VisitedPtrMap) {
 	originalType := original.Type()
 	cloneMethod, cloneMethodFound := originalType.MethodByName(`DeepCopy`)
 	kind := original.Kind()
 
 	// Prevent cycles
 	if kind == reflect.Ptr && !original.IsNil() {
 		ptr := original.Pointer()
-		if visited[ptr] {
-			panic(fmt.Errorf("deepcopy cycle detected\neach pointer can be used only once\nsteps: %s", steps.String()))
+		if v, found := visitedPtr[ptr]; found {
+			clone.Set(*v)
+			return
+		} else {
+			visitedPtr[ptr] = &clone
```

Comment:
Funguje to takto, ak sme dany pointer uz "videli", tak ho neklonujeme znova, ale pouzijeme hodnotu, ktoru sme vygenerovali predtym.

Teda ak v podovnej datovej strukture bol ten isty pointer 3x, tak aj v kopii bude pointer na kopiu 3x.